### DISCUSSION TITLE
Invalid input field does not trigger onChange on first keystroke

### DISCUSSION THREAD

**Discussion by jaymehta-twilio** [react-hook-form/react-hook-form#5529] [Q&A] [answered]
Hi,

I've wrapped the Input of our internal library to work with react-hook-form. Here's the wrapper component:

```
import { Input as $Input, InputProps as $InputProps } from '@twilio-paste/core/input';
import { FieldValues } from 'react-hook-form';

export function Input<TKeys extends FieldValues>(
  props: Omit<$InputProps, 'ref'> & {
    registerRef: React.Ref<HTMLInputElement>;
    name: keyof TKeys;
  },
): React.ReactElement {
  const { registerRef: ref, ...rest } = props;
  return <$Input ref={ref} {...rest} />;
}

Input.displayName = 'Input';
```

Here's the usage:

```
<Label htmlFor="external_ID" required>
            {createSinkLabels.externalIdFieldLabel}
          </Label>
          <Input
            aria-describedby={createSinkLabels.externalIdFieldLabel}
            id="External_ID"
            name="External_ID"
            registerRef={register({ required: true })}
            type="text"
            data-testid="ExternalId-field"
            hasError={errors.External_ID ? true : undefined}
            value={localExtId || ""}
            onChange={extIdInputChangeHandler}
          />
          {errors.External_ID ? (
            <HelpText variant="error" data-testid="extId-error-helptext">
              {createSinkLabels.externalIdFieldHelpText}
            </HelpText>
          ) : (
            <HelpText data-testid="extId-field-helptext">
              {createSinkLabels.externalIdFieldHelpText}
            </HelpText>
          )}
```

This works just fine. But, as soon as the field is in invalid state, the onChange doesn't trigger on the first keystroke. However, it does on the 2nd keystroke. At first I think it could be due to focus and tried `autoFocus: {false}` but it doesn't work. Is there something obvious I am missing here ? Thanks for the help in advance. 🙏🏽

<img width="917" alt="Screen Shot 2021-06-08 at 2 10 59 PM" src="https://user-images.githubusercontent.com/52717786/121258713-15070b00-c864-11eb-9055-cc388dab1bcb.png">

**Comment by bluebill1049** [comment]
can you share a reproducible codesandbox? need more context, such as version and etc...

**Reply by jaymehta-twilio** [reply]
@bluebill1049 Here's my CSB: https://codesandbox.io/s/condescending-pasteur-exmun?file=/src/index.tsx
The version we use is `6.13.1`

**Comment by jaymehta-twilio** [comment, marked as answer]
I found the solution. This was happening because I was using uncontrolled inputs with local state, which was causing the issue. Thank you 👍
